compiler/include/ForallStmt.h

```diff
@@ -69,24 +69,38 @@ class ForallOptimizationInfo {
     // forall loop statement //
 ///////////////////////////////////
 
-class ForallStmt final : public Stmt
+struct ShadowVarLoopInterface {
```

Comment:
Can we find a better place to live for this?

Also consider `LoopWithShadowVars` as a name alternative.

Reply:
How about renaming it to LoopWithShadowVarsInterface? It's wordy but I like having 'Interface' in the name just so its obvious that this isn't a new node the AST or something like that.

And how about I make a new `LoopWithShadowVarsInterface.h` file for it? Alternatively I could put it in `ForLoop.h` or `intents.h` but neither of those seem like a good place to have it either.